test/computDYsplitting_HT.cpp

```diff
@@ -0,0 +1,181 @@
+#include "bigTree.h" 
+#include "smallTree.h"
+#include "OfflineProducerHelper.h"
+#include "PUReweight.h"
+#include "triggerReader.h"
+#include "bJetRegrVars.h"
+#include "bTagSF.h"
+
+using namespace std ;
+
+#define DEBUG false
+
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ----
+
+bool CheckBit (int number, int bitpos)
+{
+    bool res = number & (1 << bitpos);
+    return res;
+}
+
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ----
+string printFlag (int flag, int max = 16)
+{
+    string s = "";
+    for (int i = max; i >=0; i--)
+    {
+        if (CheckBit (flag, i)) s+="1";
+        else s+="0";
+    }
+    return s;
+}
+
+// ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- -
+// open input txt file and append all the files it contains to TChain
+void appendFromFileList (TChain* chain, TString filename)
+{
+    std::ifstream infile(filename.Data());
+    std::string line;
+    while (std::getline(infile, line))
+    {
+        line = line.substr(0, line.find("#", 0)); // remove comments introduced by #
+        while (line.find(" ") != std::string::npos) line = line.erase(line.find(" "), 1); // remove white spaces
+        while (line.find("\n") != std::string::npos) line = line.erase(line.find("\n"), 1); // remove new line characters
+        while (line.find("\r") != std::string::npos) line = line.erase(line.find("\r"), 1); // remove carriage return characters
+        if (!line.empty()) // skip empty lines
+            chain->Add(line.c_str());
+    }
+    return;
+}
+
+int main(int argc, char** argv)
+{
+    if (argc < 2)
+    {
+        cout << "** error of inputs . Usage:" << endl;
+        cout << "computeDYsplitting <inputFileList> <OPTIONAL : outputFileName>" << endl;
+        return 1;
+    }
+
+    string inputFile = argv[1];
+    string outputFileName = "histos_DYCount.root" ;
+    if (argc > 2) outputFileName = argv[2];
+
+    cout << "** INFO: using filelist : " << inputFile << endl;
+
+    // input and output setup
+    // ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ---- ----
+    TChain * bigChain = new TChain ("HTauTauTree/HTauTauTree") ;
+    appendFromFileList (bigChain, inputFile);
+    bigTree theBigTree (bigChain) ;
+
+    int totEvents = 0;
+    int allEvts[5][5][9]; // njet,nb,ht
+    for (int i = 0; i < 5; i++)
+        for (int j = 0; j < 5; j++)
+            for (int k = 0; k < 9; k++)
+                allEvts[i][j][k] = 0;
+		
+    float xbins[6]  = {0.0, 1.0, 2.0, 3.0, 4.0, 5.0};
+    float ybins[6]  = {0.0, 1.0, 2.0, 3.0, 4.0, 5.0};
+    float zbins[10] = {0.0,70.0,100.0,200.0,400.0,600.0,800.0,1200.0,2500.0,10000.0};
+	    
+    TFile* fOut = new TFile (outputFileName.c_str(), "recreate");
+    TH3F* h_nJets_nBs    = new TH3F ("h_nJets_nBs"   , ";nJets;nBs;ht", 5, xbins, 5, ybins, 9, zbins);
+    TH3I* hINT_nJets_nBs = new TH3I ("hINT_nJets_nBs", ";nJets;nBs;ht", 5, xbins, 5, ybins, 9, zbins);
```

Comment:
can you canghe the names of the TH3 histograms here? lsomething like `h_nJets_nBs_HT` and `hINT_nJets_nBs_HT`

Reply:
Done!